### ISSUE TITLE
[Bug]: When running full test, service crashes

### ISSUE THREAD

**Issue by Ognimalf** [infiniflow/infinity#665]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

1af0c66

### Actual behavior

During running the full tests, the service will crash occasionally.
The backtrace collected by gdb is below:

```
Thread 21 "infinity" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffde5416c0 (LWP 134286)]
0x0000555555f1d6ab in infinity::EqualTo@stl<unsigned long>::operator()(unsigned long const&, unsigned long const&) const (this=0x5130000005d0, left=@0x7fffdd3cbda0: 13741, 
    right=<error reading variable: Cannot access memory at address 0x8>) at /home/xwg/dev/infinity/src/common/stl.cppm:245
245             bool operator()(const T &left, const T &right) const { return left == right; }
(gdb) bt
#0  0x0000555555f1d6ab in infinity::EqualTo@stl<unsigned long>::operator()(unsigned long const&, unsigned long const&) const (this=0x5130000005d0, left=@0x7fffdd3cbda0: 13741, 
    right=<error reading variable: Cannot access memory at address 0x8>) at /home/xwg/dev/infinity/src/common/stl.cppm:245
#1  0x000055555672b1c4 in std::__detail::_Hashtable_base<unsigned long, std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, std::__detail::_Select1st, infinity::EqualTo@stl<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<false, false, true> >::_M_key_equals(unsigned long const&, std::__detail::_Hash_node_value<std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, false> const&) const (this=0x5130000005d0, __k=@0x7fffdd3cbda0: 13741, __n=...)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/hashtable_policy.h:1724
#2  0x000055555672b013 in std::__detail::_Hashtable_base<unsigned long, std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, std::__detail::_Select1st, infinity::EqualTo@stl<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<false, false, true> >::_M_equals(unsigned long const&, unsigned long, std::__detail::_Hash_node_value<std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, false> const&) const (this=0x5130000005d0, __k=@0x7fffdd3cbda0: 13741, __c=13741, __n=...)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/hashtable_policy.h:1743
#3  0x000055555672aefa in std::_Hashtable<unsigned long, std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, std::allocator<std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> > >, std::__detail::_Select1st, infinity::EqualTo@stl<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_find_before_node(unsigned long, unsigned long const&, unsigned long) const (this=0x5130000005d0, __bkt=0, __k=@0x7fffdd3cbda0: 13741, 
    __code=13741) at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/hashtable.h:1946
#4  0x000055555672a5fd in std::_Hashtable<unsigned long, std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, std::allocator<std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> > >, std::__detail::_Select1st, infinity::EqualTo@stl<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_find_node(unsigned long, unsigned long const&, unsigned long) const (this=0x5130000005d0, __bkt=0, __key=@0x7fffdd3cbda0: 13741, __c=13741)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/hashtable.h:815
#5  0x000055555672db8f in std::_Hashtable<unsigned long, std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> >, std::allocator<std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> > >, std::__detail::_Select1st, infinity::EqualTo@stl<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::find(unsigned long const&) (this=0x5130000005d0, __k=@0x7fffdd3cbda0: 13741)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/hashtable.h:1664
#6  0x0000555556727edb in std::unordered_map<unsigned long, std::shared_ptr<infinity::Txn@txn>, std::hash<unsigned long>, infinity::EqualTo@stl<unsigned long>, std::allocator<std::pair<unsigned long const, std::shared_ptr<infinity::Txn@txn> > > >::find(unsigned long const&) (this=0x5130000005d0, __x=@0x7fffdd3cbda0: 13741)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/unordered_map.h:876
#7  0x00005555567250f0 in infinity::TxnManager@txn_manager::GetMinUncommitTs() (this=0x513000000580) at /home/xwg/dev/infinity/src/storage/txn/txn_manager.cpp:167
#8  0x000055555704b1fb in infinity::CleanupPeriodicTrigger@periodic_trigger::Trigger() (this=0x5060000029c0) at /home/xwg/dev/infinity/src/storage/bg_task/periodic_trigger.cpp:31
#9  0x0000555557128b77 in infinity::PeriodicTriggerThread@periodic_trigger_thread::Run() (this=0x50400000f5d0) at /home/xwg/dev/infinity/src/storage/periodic_trigger_thread.cpp:30
#10 0x0000555556e6ad9a in infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0::operator()() const (this=0x502000000418)
    at /home/xwg/dev/infinity/src/storage/periodic_trigger_thread.cppm:42
#11 0x0000555556e6ad55 in std::__invoke_impl<void, infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0>(std::__invoke_other, infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0&&) (__f=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/invoke.h:61
#12 0x0000555556e6ad15 in std::__invoke<infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0>(infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0&&) (__fn=...)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/invoke.h:96
#13 0x0000555556e6aced in std::thread::_Invoker<std::tuple<infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0x502000000418)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_thread.h:292
#14 0x0000555556e6acc5 in std::thread::_Invoker<std::tuple<infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0> >::operator()() (this=0x502000000418)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_thread.h:299
#15 0x0000555556e6ac09 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<infinity::PeriodicTriggerThread@periodic_trigger_thread::Start()::$_0> > >::_M_run() (this=0x502000000410)
    at /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_thread.h:244
#16 0x00005555592c1a83 in execute_native_thread_routine ()
#17 0x0000555555b00ccb in asan_thread_start(void*) ()
#18 0x00007ffff7c8f6ba in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:444
#19 0x00007ffff7d1e120 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

### Steps to reproduce

```Markdown
This bug occasionally occurs during multiple runs of all tests. Just run `python tools/sqllogictest.py`
```

### Additional information

According to the call stack, this bug may be related to this commit. 1af0c66

**Comment by small-turtle-1** [comment]
Fixed in pr #667

**Comment by Ognimalf** [comment]
Up to today the error doesn't appear again, looks fixed.
